### ISSUE TITLE
Injective and Isomorphic type classes?

### ISSUE THREAD

**Issue by DanielGronau** [highj/highj#76]
I had some older code for this. Is this useful or not?

**Comment by clinuxrulz** [comment]
Yes I do believe this is useful. E.g. you may make a combined ParserPrettyPrinter out of combinators resulting in something with a type param made up of products and coproducts, and then you need to go from that type to the type your interested in working with (via Iso).

**Comment by DanielGronau** [comment]
I added `Function1` as weaker version of `Injective1`, seems to have some interesting applications

**Comment by clinuxrulz** [comment]
I do not see ```Function1``` in the source. (not yet pushed?).
However would ```Function1``` be the same as ```NF```?
( https://github.com/highj/highj/blob/master/src/main/java/org/highj/function/NF.java )

**Comment by DanielGronau** [comment]
You are right, it is the same. I was all the time thinking "shouldn't we have this already?", but missed NF. I'll throw Function1 away.
